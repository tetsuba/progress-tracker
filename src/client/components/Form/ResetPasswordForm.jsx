import React, { useState } from 'react';

import { Button, Form, Row } from 'react-bootstrap';
import PasswordStrength from "../PasswordStrength/PasswordStrength";
import {handleInputChange, passwordMatchError, passwordsMatched} from "./form-utils";

const ResetPasswordForm = ({ userId }) => {
    const [inputs, setInputs] = useState({password1: '', password2: ''});
    const [error] = useState({message: ''});

    console.log('userId:', userId)

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputs.password1 !== inputs.password2) {
            return;
        }

        // sendPasswordResetConfirmation({ variables: { input: inputs } })
        //     .then(({ data }) => {
        //         setSuccess(data.sendPasswordResetConfirmation.confirmation)
        //     })
        //     .catch((error) => {
        //         const message = error.graphQLErrors[0].message;
        //         setError({ message })
        //     })
    };

    return (
        <Row>
            <Form
                className="w-100"
                onSubmit={(e) => handleSubmit(e)}
            >
                <Form.Group controlId="formPassword1">
                    <Form.Control
                        required
                        type="password"
                        placeholder="Password"
                        name="password1"
                        onChange={(e) => handleInputChange(e, setInputs)}
                        value={inputs.password1}
                    />
                    <Form.Control.Feedback type="invalid">
                        { error.message }
                    </Form.Control.Feedback>
                    <PasswordStrength password={inputs.password1}  />
                </Form.Group>
                <Form.Group controlId="formPassword2">
                    <Form.Control
                        required
                        type="password"
                        placeholder="Password"
                        name="password2"
                        onChange={(e) => handleInputChange(e, setInputs)}
                        value={inputs.password2}
                        isInvalid={passwordMatchError(inputs)}
                    />
                    <Form.Control.Feedback type="invalid">
                        Password does not match!!!
                    </Form.Control.Feedback>
                    <Form.Text className="text-muted">
                        { `Matching password ${inputs.password2.length} of ${inputs.password1.length }` }
                    </Form.Text>
                </Form.Group>
                <Button variant="primary" type="submit" disabled={passwordsMatched(inputs)}>
                    Reset Password
                </Button>
            </Form>
        </Row>
    )
};

export default ResetPasswordForm
