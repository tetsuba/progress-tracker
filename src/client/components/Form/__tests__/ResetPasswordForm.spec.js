import { testRenderer } from '../../../../test/testHelper'

// COMPONENTS
import ResetPasswordForm from '../ResetPasswordForm'

describe('<ResetPasswordForm>', () => {
  const props = {
    userId: '1234567890',
    resetPassword: jest.fn(),
  }

  function updateInput(wrapper, name, value) {
    wrapper.find({ name }).get(0).props.onChange({
      persist: jest.fn(),
      target: { name, value },
    })
  }

  describe('@Render', () => {
    it('should render reset password form', () => {
      const wrapper = testRenderer(ResetPasswordForm, props)
      expect(wrapper).toMatchSnapshot()
    })

    it('should disable button if passwords do not match', () => {
      const wrapper = testRenderer(ResetPasswordForm, props)
      updateInput(wrapper, 'newPassword', '123456')
      updateInput(wrapper, 'confirmPassword', '123457')
      expect(wrapper.find({ type: 'submit' }).prop('disabled')).toBeTruthy()
    })
  })

  describe('@Events', () => {
    it('should update newPassword input with event onChange', async () => {
      const wrapper = testRenderer(ResetPasswordForm, props)
      const name = 'newPassword'
      const value = '123456'
      updateInput(wrapper, name, value)
      expect(wrapper.find({ name }).get(0).props.value).toEqual(value)
    })

    it('should update confirmPassword input with event onChange', async () => {
      const wrapper = testRenderer(ResetPasswordForm, [], props)
      const name = 'confirmPassword'
      const value = '123456'
      updateInput(wrapper, name, value)
      expect(wrapper.find({ name }).get(0).props.value).toEqual(value)
    })

    it('should call event onSubmit', () => {
      const wrapper = testRenderer(ResetPasswordForm, props)
      const id = '#ResetPasswordForm'
      wrapper.find(id).simulate('submit', { preventDefault: jest.fn() })
      expect(props.resetPassword).toHaveBeenCalledTimes(1)
    })
  })

  describe('@Error', () => {
    it('should render an error message if passwords do not match', async () => {
      const wrapper = testRenderer(ResetPasswordForm, props)
      updateInput(wrapper, 'newPassword', '123456')
      updateInput(wrapper, 'confirmPassword', '123457')
      expect(wrapper.find({ type: 'invalid' }).prop('children')).toEqual(
        'Passwords do not match!!!'
      )
    })
  })
})
