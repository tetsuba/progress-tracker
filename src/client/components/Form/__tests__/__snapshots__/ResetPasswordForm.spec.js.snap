// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordForm> @Render should render reset password form 1`] = `
<Fragment>
  <Row
    className="mt-5 mb-1"
    noGutters={false}
  >
    <h1>
      Reset Password
    </h1>
  </Row>
  <Row
    noGutters={false}
  >
    <Form
      className="w-100"
      id="ResetPasswordForm"
      inline={false}
      onSubmit={[Function]}
    >
      <FormGroup
        className="mb-3"
        controlId="newPassword"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          New password
        </FormLabel>
        <FormControl
          name="newPassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <PasswordStrength
          password=""
        />
      </FormGroup>
      <FormGroup
        className="mb-3"
        controlId="confirmPassword"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Confirm password
        </FormLabel>
        <FormControl
          isInvalid={false}
          name="confirmPassword"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
        <Feedback
          type="invalid"
        />
        <FormText
          className="text-muted"
        >
          Matching password 0 of 0
        </FormText>
      </FormGroup>
      <Button
        active={false}
        disabled={false}
        id="ResetPasswordSubmit"
        type="submit"
        variant="primary"
      >
        Reset Password
      </Button>
    </Form>
  </Row>
</Fragment>
`;
